src/component/DefaultTooltipContent.tsx

```diff
@@ -19,7 +19,7 @@ export type Formatter<TValue extends ValueType, TName extends NameType> = (
   item: Payload<TValue, TName>,
   index: number,
   payload: Array<Payload<TValue, TName>>,
-) => [TValue, TName] | TValue;
+) => [React.ReactNode, React.ReactNode] | React.ReactNode;
```

Comment:
`[TValue, TName] | React.ReactNode;` would be more appropriate no? Based on the conditionals below, if formatter returns an array the name will not be rendered if it is not a string or a number. If someone returns a react element in the array `[<div>{name}</div>, <div>{value}</div>]` only value will be rendered

Reply:
Since the value always rendered regardless of the type of name, isn't better to make it:

`[React.ReactNode, TName] | React.ReactNode` so the value type will be consistent?

Reply:
I guess, but if someone is providing a ReactNode imo they would just use only a react node and do

```
<>
  {myName}: {myValue}
</>
```

instead of

`[<div>{someValue}</div>, 'some name'`]`

Reply:
but the suggested change is also fine, it removes my concern

Reply:
The motivation for changing this was raised in #2976 you don't need to be bound to the input type, and while the `TName` keeps the name in check, the value can be more flexible since it can be anything from users perspective, but here the types keep you from doing passing back anything outside of inputs scope, I say ReactNode because the value is rendered as its, it enables people to do whatever they need since it doesn't concern recharts what the rendered value is.